libs/radio/sim/radio.ts

```diff
@@ -33,7 +33,17 @@ namespace pxsim.radio {
     export function readRawPacket() {
         const state = pxsim.getRadioState();
         const packet = state.datagram.recv();
-        return new RefBuffer(packet.payload.bufferData)
+        const buf = packet.payload.bufferData;
+        const n = buf.length;
+        if (!n)
+            return undefined;
+
+        const rbuf = BufferMethods.createBuffer(n + 4);
```

Comment:
are these 4 bytes at the end stripped somewhere?

Reply:
they stay around, the typescript parser knows about those

Reply:
https://github.com/microsoft/pxt-common-packages/blob/master/libs/radio/radio.ts#L182

Reply:
We need to store the RSSI per packet; not globally otherwise it is incorrect.